internal/proxy/task_query.go

```diff
@@ -309,17 +308,38 @@ func (t *queryTask) Execute(ctx context.Context) error {
 	log := log.Ctx(ctx)
 
 	executeQuery := func(withCache bool) error {
-		shards, err := globalMetaCache.GetShards(ctx, withCache, t.collectionName)
-		if err != nil {
-			return err
+		query := func() error {
```

Comment:
if we already support retry then we should clean this code.
the first request always retried, if err then based on error type we decide whether we should clean cache.